bisection.py

```diff
@@ -0,0 +1,373 @@
+        # Return the result json file
+        output_dir = os.path.join(self.workdir, commit.sha)
+        # If the directory exists, delete its contents
+        if os.path.exists(output_dir):
+            assert os.path.isdir(output_dir), "Must specify output directory: {output_dir}"
+            filelist = [ f for f in os.listdir(output_dir) ]
+            for f in filelist:
+                os.remove(os.path.join(output_dir, f))
+        else:
+            os.mkdir(output_dir)
+        command = str()
+        if self.bmfilter:
+            command = f"bash .github/scripts/run-nodocker.sh {output_dir} \"{self.bmfilter}\" &> {output_dir}/benchmark.log"
+        else:
+            command = f"bash .github/scripts/run-nodocker.sh {output_dir} &> {output_dir}/benchmark.log"
+        try:
+            subprocess.check_call(command, cwd=self.srcpath, shell=True, timeout=self.timelimit * 60)
+        except subprocess.TimeoutExpired:
+            print(f"Benchmark timeout for {commit.sha}. Returning zero value.")
+            return output_dir
+        return output_dir
+
+    def compute(self, result_dir: str) -> float:
+        filelist = [ f for f in os.listdir(result_dir) if f.endswith(".json") ]
+        if len(filelist) == 0:
+            print(f"Empty directory or json file in {result_dir}. Return zero score.")
+            return 0.0
+        # benchmark data file
+        data_file = os.path.join(result_dir, filelist[0])
+        if os.stat(data_file).st_size == 0:
+            print(f"Empty json file {filelist[0]} in {result_dir}. Return zero score.")
+            return 0.0
+        # configuration
+        config_file = os.path.join(self.srcpath, TORCHBENCH_SCORE_CONFIG)
+        with open(config_file) as cfg_file:
+            config = yaml.full_load(cfg_file)
+        with open(data_file) as dfile:
+            data = json.load(dfile)
+        return compute_score(config, data)
+    
+    def get_score(self, commit: Commit) -> float:
+        # Score is cached
+        if commit.score is not None:
+            return commit.score
+        # Build pytorch and its dependencies
+        self.torch_src.build(commit)
+        # Run benchmark
+        print(f"Running TorchBench for commit: {commit.sha} ...", end="", flush=True)
+        result_dir = self.run_benchmark(commit)
+        commit.score = self.compute(result_dir)
+        print(f" score: {commit.score}")
+        self.torch_src.cleanup(commit)
+        return commit.score
+        
+class TorchBenchBisection:
+    start: str
+    end: str
+    workdir: str
+    threshold: int
+    bisectq: List[Tuple[Commit, Commit]]
+    result: List[Tuple[Commit, Commit]]
+    torch_src: TorchSource
+    bench: TorchBench
+    output_json: str
+
+    def __init__(self,
+                 workdir: str,
+                 torch_src: str,
+                 bench_src: str,
+                 start: str,
+                 end: str,
+                 threshold: int,
```

Comment:
maybe should be a float, im sure int is good enough for now

Reply:
Fixed by converting to float